workflow/controller/operator.go

```
@@ -901,11 +907,21 @@ func (woc *wfOperationCtx) podReconciliation(ctx context.Context) error {
 				woc.updated = true
 			}
 			node := woc.wf.Status.Nodes[pod.ObjectMeta.Name]
+			match := true
+			if woc.execWf.Spec.PodGC != nil && woc.execWf.Spec.PodGC.LabelSelector != nil {
+				labelsMap, err := metav1.LabelSelectorAsMap(woc.execWf.Spec.PodGC.LabelSelector)
+				if err != nil {
+					woc.log.Warnf("Failed to parse label selector for pod GC: %s", woc.execWf.Spec.PodGC.LabelSelector)
```

ignore the error but don't continue?